### PR TITLE
Provide default type for schema fields 

### LINKED ISSUE #3599: Provide default type for schema fields
For the schemas: time should default to duration, size should default to size

### PR DESCRIPTION
**Pull request by jzwang43**
### Reasons for making this change

Provide default types for some schema paths so it's easier to use.

### Related issues

#3599 

### Screenshots

test
### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js [hunk 1]
```diff
@@ -128,13 +130,38 @@ class SchemaItem extends React.Component<{
...
         this.setState({ rows: [...copiedRows] });
     };
 
+    autofillTypes = (row) => {
+        // Do not fill if how to render is defined.
+        if (row['post-processor'] !== null) {
+            return row;
+        }
+
+        switch (row['generalized-path'].toLowerCase()) {
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: Can we instead use a dictionary with a name like `DEFAULT_POST_PROCESSOR` so it's easier to configure in the future?

Comment (jzwang43) [reply frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: Thanks, I added it to constants.

File: frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js [hunk 2]
```diff
@@ -128,13 +131,29 @@ class SchemaItem extends React.Component<{
...
+            this.autofillTypes({
+                ...rows[idx],
+                [key]: e.target.value.replace(/\n/g, ' ').replace("'", ''),
+            }),
+        );
         this.setState({ rows: [...copiedRows] });
     };
 
+    autofillTypes = (row) => {
+        // Do not fill if how to render is defined.
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: I don't understand this comment.

Can you also explain in the comment what `null` means? does it mean that it needs to be autofilled?

Comment (jzwang43) [reply frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: Yeah `null` means the type has never been auto-filled so we can go ahead and auto-fill it. Let me update the comment so it's more clear.

File: frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js [hunk 3]
```diff
@@ -128,13 +131,30 @@ class SchemaItem extends React.Component<{
...
+            this.autofillTypes({
+                ...rows[idx],
+                [key]: e.target.value.replace(/\n/g, ' ').replace("'", ''),
+            }),
+        );
         this.setState({ rows: [...copiedRows] });
     };
 
+    autofillTypes = (row) => {
+        // Only autofill if how to render is null.
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]:
```suggestion
        // Only autofill if the post-processor is null.
```

File: frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js [hunk 4]
```diff
@@ -116,7 +117,9 @@ class SchemaItem extends React.Component<{
         const schemaHeaders = schemaItem.header;
         let newRow = { from_schema_name: schemaItem.schema_name };
         schemaHeaders.forEach((header) => {
-            newRow[header] = '';
+            // Set post-processor to null for new rows so we can tell if a row has been autofilled.
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]:
```suggestion
            // Set post-processor to null for new, unedited rows to indicate that the row can be autofilled.
```

File: frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js [hunk 5]
```diff
@@ -116,7 +117,9 @@ class SchemaItem extends React.Component<{
...
+            // This also makes new row default values consistent with the first row.
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: What does this mean?

Comment (jzwang43) [reply frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: We need to set the default post-processor to null so we can tell the difference between initialized value and when user manually deleted the input for post-processor.

Comment (epicfaace) [reply frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: > We need to set the default post-processor to null so we can tell the difference between initialized value and when user manually deleted the input for post-processor.

I think that's what the first comment ("Set post-processor to null for new, unedited rows to indicate that the row can be autofilled.") means. I don't understand what the second comment ("This also makes new row default values consistent with the first row.") means. Can you modify it to make it clearer?

Comment (jzwang43) [reply frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js]: > 
> 
> > We need to set the default post-processor to null so we can tell the difference between initialized value and when user manually deleted the input for post-processor.
> 
> I think that's what the first comment ("Set post-processor to null for new, unedited rows to indicate that the row can be autofilled.") means. I don't understand what the second comment ("This also makes new row default values consistent with the first row.") means. Can you modify it to make it clearer?

The old code sometime sets post-processor to `null` and sometimes `''` and it's confusing.
The second line was referring to this issue. But I this the first line of the comment should be enough to explain why `null` is used so I removed the second line.

**Review by epicfaace — APPROVED** [review]

**Pull request merged**